src/Core/src/Hosting/Internal/MauiServiceProvider.cs

```diff
@@ -97,52 +178,77 @@ public MauiServiceProvider(IMauiServiceCollection collection, bool constructorIn
 
 		object? CreateInstance(Type implementationType)
 		{
-			(ConstructorInfo Constructor, ParameterInfo[] Parameters) match = default;
+			// get constructors ordered by parameter count
```

Comment:
Could we include some tests related with this?

Reply:
Added.